### DISCUSSION TITLE
❓➡❕ Dedicated Q&A

### DISCUSSION THREAD

**Discussion by raphaelschmitz00** [stride3d/stride#1837] [Ideas]
I've only been here a week or so, but I have already seen some questions multiple times.

Don't get me wrong, the people answering those have been very friendly and helpful,
but I fear they might get annoyed over time. Especially with the whole Unity situation,
where we now have a small team of people who knew Stride well before, and a big influx
of people starting from zero.

A dedicated Q&A site (a site like stackoverflow) could help with that.
The main advantage, TBH, would probably that it's googleable then.
I know there's a search function on Discord, but it's not tooo great and that's not
where people seem to write their questions.

The disadvantage is that it takes more effort for question askers, than just having the chat for it.
That's a double edged sword though IMHO:
A) Up until which frequency do we want to provide the service of answering recurring questions there?
B) If they found it in google, they might not even have asked it in chat in the first place.

BUUUUT this doesn't make sense if the people who are currently answering questions
wouldn't see it as an improvement, or otherwise not use it.

So, let me know: Are you actually even annoyed by repeating questions?
More importantly, would you prefer going to another website to answer them once and for all?

---

PS
This has a clear distinction with [the currently ongoing documentation efforts](https://github.com/stride3d/stride/discussions/1745).
It would definitely not replace those. However, writing documentation like that, it's
making a guess about what somebody else _probably_ wants to know.
A Q&A is answering something we already know they want to know - but it's also less general.
(These answers can work as a starting point for articles in the documentation.)

PPS
We might want to ask the guys at [Codidact](https://codidact.com/) if they would host a "community" for us.
The whole Codidact project was started after StackExchange pulled some evil corporation moves,
so there's some kind of overlap with the Unity situation 🤣

That one, it's open source and we _could_ even host ourselves...
but more maintenance work for us, meh, not a fan personally.
Other services are also possible, of course.

**Comment by YuechenLi-VG** [comment]
Stride doesn't have a forum right? Self hosting something like a Lemmy instance would be good to get that initial userbase from federation, and imo the reddit like structure makes searching for information easier than that of traditional forums. 

https://github.com/LemmyNet/lemmy

**Comment by VaclavElias** [comment]
Hi @raphaelschmitz00,

Thank you for your comments and suggestions. You're absolutely right in your observations. However, we currently lack the manpower to maintain or check multiple platforms for **Q&A**. We have decided to focus our efforts on **Discord** and **GitHub**.

Many questions we receive are technical and often require responses from experienced or core developers. These developers have limited time to contribute to the Stride repository, let alone check other platforms.

In the past, we made [Stride Forums](https://forums.stride3d.net/) read-only due to decreased visits and removed our [Disqus forum](https://disqus.com/home/forum/stride3d/) from our docs for similar reasons, including unanswered questions. This one seems also not used https://gamedev.stackexchange.com/questions/tagged/stride.

Discord serves us well. We receive notifications there for any GitHub Issue and comment in the Stride, Stride Web, and Stride Docs repositories. This allows us to manage our community and contributions efficiently.

Regarding repetitive questions, we're aware this is an issue and are taking steps to address it. We recently launched a new website and are working on adding articles like [this one](https://www.stride3d.net/blog/closer-look-private-members/) to cover hot topics in Stride.

We've also just rolled out a new docs website (today! still working on it!, that's why we are silent) featuring a modern design and powered by the latest version of DocFX. Our aim is to revamp our manuals and add more content, incorporating questions from Discord where appropriate. However, this is a time-consuming process and requires volunteer effort.

While GitHub Discussions serves as a decent Q&A platform, it's not well-indexed by search engines. We're considering adding a dedicated Q&A section or some kind of form of this (most likely static) to our new docs, which would likely be better indexed.

As our community grows and we gain more volunteers, we'll be more open to exploring other platforms. Until then, we're focusing on improving Stride Docs, providing more examples, and offering detailed guides.

To summarize, our current focus is on enhancing our documentation to serve as a comprehensive resource, aiming to answer a wide range of community questions. We believe that this effort will significantly reduce the frequency of repeated inquiries. 

We prioritize answering questions, even if they are repetitive, over leaving them unanswered on any platform 🙂.

Saying that, if there is any volunteer how would like to collect Q&A which we should add to Stride Docs, please let us know!

**Reply by raphaelschmitz00** [reply]
OK makes sense, if there is no problem, then there's nothing to solve.
